### PR TITLE
AG-12847 Minify filter code

### PR DESCRIPTION
**Pull request by StephenCooper**

### CODE REVIEW

File: packages/ag-grid-community/src/filter/filterManager.ts [hunk 1]
```diff
@@ -153,26 +140,16 @@ export class FilterManager extends BeanStub implements NamedBean {
...
-    public isAdvancedFilterEnabled(): boolean {
+    public isAdvFilterEnabled(): boolean {
         return !!this.advancedFilter?.isEnabled();
     }
 
-    public isAdvancedFilterHeaderActive(): boolean {
-        return this.isAdvancedFilterEnabled() && this.advancedFilter.isHeaderActive();
-    }
-
-    public isAnyFilterPresent(): boolean {
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/filter/filterManager.ts:164]: Now sharing logic with isChildFilterPresent above.

File: packages/ag-grid-community/src/filter/floating/provided/floatingFilterTextInputService.ts [hunk 2]
```diff
@@ -5,56 +5,49 @@ import { RefPlaceholder } from '../../../widgets/component';
...
             input: listener,
             keydown: listener,
         });
     }
 
     public setEditable(editable: boolean): void {
-        this.eFloatingFilterTextInput.setDisabled(!editable);
-    }
-
-    public setAutoComplete(autoComplete: boolean | string): void {
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/filter/floating/provided/floatingFilterTextInputService.ts:33]: Need to add back

File: packages/ag-grid-community/src/filter/floating/provided/floatingFilterTextInputService.ts [hunk 3]
```diff
@@ -5,56 +5,49 @@ import { RefPlaceholder } from '../../../widgets/component';
...
+        eInput.setInputAriaLabel(ariaLabel);
 
-        if (params.autoComplete !== undefined) {
-            this.setAutoComplete(params.autoComplete);
+        if (autoComplete !== undefined) {
+            eInput.setAutoComplete(autoComplete);
         }
     }
-
-    private setAriaLabel(ariaLabel: string): void {
```
Comment (StephenCooper) [inline packages/ag-grid-community/src/filter/floating/provided/floatingFilterTextInputService.ts:57]: Just inlined above.

File: packages/ag-grid-community/src/filter/floating/provided/floatingFilterTextInputService.ts (hunk 2 above)
Comment (StephenCooper) [inline packages/ag-grid-community/src/filter/floating/provided/floatingFilterTextInputService.ts:33]: Can't see any external uses so have inlined below.

File: packages/ag-grid-community/src/filter/floating/provided/simpleFloatingFilter.ts [hunk 4]
```diff
@@ -17,57 +15,30 @@ import type { SimpleFilterModelFormatter } from '../../provided/simpleFilterMode
...
-    private readOnly: boolean;
-
-    protected getDefaultDebounceMs(): number {
-        return 0;
-    }
+    protected readOnly: boolean;
+    protected defaultDebounceMs: number = 0;
 
     // this is a user component, and IComponent has "public destroy()" as part of the interface.
     // so we need to override destroy() just to make the method public.
```
Comment (peterjrreynolds) [inline packages/ag-grid-community/src/filter/floating/provided/simpleFloatingFilter.ts]: We made `destroy` public on `BeanStub`, so can remove this now.

File: packages/ag-grid-community/src/filter/provided/providedFilter.ts [hunk 5]
```diff
@@ -28,7 +26,7 @@ import type { IProvidedFilter, ProvidedFilterParams } from './iProvidedFilter';
  */
 export abstract class ProvidedFilter<M, V> extends Component implements IProvidedFilter, IFilterComp {
     // each level in the hierarchy will save params with the appropriate type for that level.
-    private providedFilterParams: ProvidedFilterParams;
+    private params: ProvidedFilterParams;
```
Comment (peterjrreynolds) [inline packages/ag-grid-community/src/filter/provided/providedFilter.ts:29]: If we added this as a generic, then could remove the other params objects. Might be easier to do as a separate PR.

File: packages/ag-grid-community/src/filter/provided/simpleFilter.ts [hunk 6]
```diff
@@ -542,17 +537,28 @@ export abstract class SimpleFilter<M extends ISimpleFilterModel, V, E = AgInputT
...
+        removeComponents(eTypes, startPosition, deleteCount);
+        removeElements(eConditionBodies, startPosition, deleteCount);
+        removeValueElements(startPosition, deleteCount);
         const joinOperatorIndex = Math.max(startPosition - 1, 0);
-        this.removeElements(this.eJoinOperatorPanels, joinOperatorIndex, deleteCount);
-        this.removeComponents(this.eJoinOperatorsAnd, joinOperatorIndex, deleteCount);
-        this.removeComponents(this.eJoinOperatorsOr, joinOperatorIndex, deleteCount);
+        removeElements(eJoinOperatorPanels, joinOperatorIndex, deleteCount);
+        removeComponents(eJoinOperatorsAnd, joinOperatorIndex, deleteCount);
+        removeComponents(eJoinOperatorsOr, joinOperatorIndex, deleteCount);
```
Comment (peterjrreynolds) [inline packages/ag-grid-community/src/filter/provided/simpleFilter.ts]: Don't destructure methods

File: packages/ag-grid-community/src/filter/provided/providedFilter.ts (hunk 5 above)
Comment (StephenCooper) [reply packages/ag-grid-community/src/filter/provided/providedFilter.ts:29]: Yeah, out of scope of this one :)

**Review by peterjrreynolds — APPROVED** [review]
